src/libcore/result.rs

```diff
@@ -1214,6 +1214,38 @@ impl<T, E> Result<Option<T>, E> {
     }
 }
 
+impl<T, E> Result<Result<T, E>, E> {
+    /// Converts from `Result<Result<T, E>, E>` to `Result<T, E>`
```

Comment:
I think the documentation could be reworded a bit, to emphasize what happens if one of the results is an `Err` variant (which wasn't really needed for `Option<T>::flatten`). It should be noted that, in Rustdoc, you don't directly see the type of `self` when looking at the definition of a method.